### PR TITLE
fix(osc): set kubelet --fail-cgroupv1=false for k8s [1.35, 1.38)

### LINKED ISSUE #339: Shoot creation failing with K8s 1.35 due to cgroup v1 incompatibility
SUSE-chost shoots with image 15.7.20260323 fail because kubelet refuses to start:

- failed to validate kubelet configuration: kubelet is configured to not run on a host using cgroup v1. cgroup v1 support is unsupported and will be removed in a future release

Root cause: Kubernetes 1.35 changed the default configuration to disable cgroup v1 support for kubelet. The SUSE-chost OS image still runs cgroup v1, causing kubelet startup to fail. Tests pass with K8s 1.34.

### PR DESCRIPTION
**Pull request by Vincinator**
**How to categorize this PR?**

/area os
/kind enhancement
/os suse-chost

**What this PR does / why we need it**:

K8s 1.35 flips kubelet's `--fail-cgroupv1` default to `true` (KEP-5573), which breaks kubelet on cgroup v1 hosts like SUSE-CHost. Per [KEP-5573](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/5573-remove-cgroup-v1), "The removal will be done no earlier than 1.38 to maintain the k8s deprecation policy." — so from 1.38 on, the flag may be removed and passing it would cause kubelet to fail to start with an unknown-flag error.

The current latest released Kubernetes version is 1.36; 1.38 is not yet released. This PR writes `/var/lib/kubelet/extra_args` with `KUBELET_EXTRA_ARGS=--fail-cgroupv1=false` only when the shoot's Kubernetes version is in `[1.35, 1.38)`.

| K8s version | cgroup at boot | Behavior |
|---|---|---|
| `< 1.35` | v1 (SUSE-CHost) | no `extra_args` file; kubelet starts (default `--fail-cgroupv1=false`) |
| `[1.35, 1.38)` | v1 (SUSE-CHost) | writes `extra_args` with `--fail-cgroupv1=false`; kubelet starts |
| `>= 1.38` | v1 (SUSE-CHost) | no file emitted; flag may be removed in kubelet |

Note: an additional gate based on the SUSE-CHost OS image version (skip the file once the image defaults to cgroup v2) will be added in a follow-up once the first SUSE-CHost version with default cgroup v2 is known.

**Which issue(s) this PR fixes**:
Fixes #339

**Special notes for your reviewer**:

**Release note**:

```bugfix operator
For shoots running Kubernetes `[1.35, 1.38)` on SUSE-CHost, kubelet is now started with `--fail-cgroupv1=false` so it works on cgroup v1 hosts. The flag is not set on K8s `>= 1.38`, where it may be removed per KEP-5573.
```

## Summary by CodeRabbit

* **New Features**
  * Operating system configuration now includes Kubernetes release-specific behavior for versions 1.35 through 1.38, ensuring optimal compatibility and stability across diverse cluster configurations.

* **Tests**
  * Significantly expanded test coverage to validate version-specific configuration behavior, error scenarios, and edge cases across multiple Kubernetes releases.

* **Chores**
  * Updated project dependencies.

[Review Change Stack](https://app.coderabbit.ai/change-stack/gardener/gardener-extension-os-suse-chost/pull/346?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: pkg/controller/operatingsystemconfig/actuator_test.go [hunk 1]
```diff
@@ -337,29 +349,88 @@ touch /var/lib/osc/provision-osc-applied
...
+					_, _, extensionFiles, _, err := actuator.Reconcile(ctx, log, osc)
+					Expect(err).NotTo(HaveOccurred())
+
+					for _, f := range extensionFiles {
+						Expect(f.Path).NotTo(Equal("/var/lib/kubelet/extra_args"))
+					}
+				})
+			})
+
+			Context("when the shoot's Kubernetes version is >= 1.35", func() {
```
Comment (Roncossek) [inline pkg/controller/operatingsystemconfig/actuator_test.go]:
```suggestion
			Context("when the shoot's Kubernetes version is >= 1.35 and < 1.38", func() {
```

**Review by Roncossek — APPROVED** [review]
/lgtm
